addons/sale_loyalty/models/sale_order.py

```diff
@@ -0,0 +1,964 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from collections import defaultdict
+
+import itertools
+
+import random
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.fields import Command
+from odoo.tools.float_utils import float_round
+from odoo.osv import expression
+
+def _generate_random_reward_code():
+    return str(random.getrandbits(32))
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    # Contains how much points should be given to a coupon upon validating the order
+    applied_coupon_ids = fields.Many2many('loyalty.card', string="Manually Applied Coupons", copy=False)
+    code_enabled_rule_ids = fields.Many2many('loyalty.rule', string="Manually Triggered Rules", copy=False)
+    coupon_point_ids = fields.One2many('sale.order.coupon.points', 'order_id', copy=False)
+    reward_amount = fields.Float(compute='_compute_reward_total')
+
+    @api.depends('order_line')
+    def _compute_reward_total(self):
+        for order in self:
+            reward_amount = 0
+            for line in order.order_line:
+                if not line.reward_id:
+                    continue
+                if line.reward_id.reward_type != 'product':
+                    reward_amount += line.price_subtotal
+                else:
+                    # Free product are 'regular' product lines with a price_unit of 0
+                    reward_amount -= line.product_id.lst_price * line.product_uom_qty
+            order.reward_amount = reward_amount
+
+    def _get_no_effect_on_threshold_lines(self):
+        """
+        Returns the lines that have no effect on the minimum amount to reach
+        """
+        self.ensure_one()
+        lines = self.env['sale.order.line']
+        return lines
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        order = super(SaleOrder, self).copy(default)
+        reward_lines = order.order_line.filtered('is_reward_line')
+        if reward_lines:
+            reward_lines.unlink()
```

Comment:
I think that the lines shouldn't be created at all, instead of unlinking them afterwards.

Reply:
It's hard to override due to how copy's code works.
See https://github.com/odoo/odoo/blob/master/odoo/models.py#L4848